Marine energy and related classes #762

l-emele opened the pull request:
New classes:
* Subclasses of `energy`:
  * `marine thermal energy`
  * `marine current energy`
  * `marine tidal energy`
  * `marine wave energy`
* Subclasses of `energy transformation`:
  * `marine thermal energy transfer`
  * `marine current energy transformation`
  * `marine tidal energy transformation`
  * `marine wave energy transformation`
* Subclasses of `water flow`:
  * `tidal flow`
  * `wave`
  * `ocean current`
* Subclasses of `liquid water`:
  * `water body`
    * `ocean`
* Subclasses of `hydro power unit`:
  * `marine current energy converting unit`
  * `marine tidal energy converting unit`
  * `marine wave energy converting unit`
* Subclasses of `hydro powerplant`:
  * `marine current energy powerplant`
  * `marine tidal energy powerplant`
  * `marine wave energy powerplant`

Changed class:
* `river`: now a subclass of `water body`

Closes #762.

sfluegel05 commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -3733,6 +3737,248 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/768",
+    
+Class: OEO_00010107
+
+    Annotations: 
```
The term tracker item for `ocean current` is missing.

sfluegel05 commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -3733,6 +3737,248 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/768",
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/806",
+        rdfs:label "marine current energy converting unit"@en
+    
+    SubClassOf: 
```
Here an axiom `participates in some marine current energy transformation` is missing.

sfluegel05 commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -3733,6 +3737,248 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/768",
+    
+    SubClassOf: 
+        OEO_00010085,
+        <http://purl.obolibrary.org/obo/RO_0000056> some OEO_00010098
```
This should be `participates in some marine tidal energy transformation` (instead of marine current).

sfluegel05 commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -3733,6 +3737,248 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/768",
+Class: OEO_00010097
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Marine current energy is the hydro energy of moving marine water.",
```
I suggested a change to this definition later on in the discussion which probably got lost: _Marine current energy is the hydro energy of **an ocean current.**_

sfluegel05 commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -3733,6 +3737,248 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/768",
+        <http://purl.obolibrary.org/obo/IAO_0000233> "issue: https://github.com/OpenEnergyPlatform/ontology/issues/762
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/806",
+        rdfs:label "marine wave energy"@en,
+        owl:equivalentClass "Marine current energy is the hydro energy of moving marine water."
```
The definition should be _Marine current energy is the hydro energy of **a tidal flow**._
Also, use used `owl:equivalentClass` instead of `definition` here.

sfluegel05 replied:
I got this wrong the first time: We are talking about `marine **wave** energy` here, so the definition should of course be _Marine **wave** energy is ..._

l-emele replied:
Now this should be fixed.

sfluegel05 commented on src/ontology/edits/oeo-physical.omn:
```diff
@@ -3733,6 +3737,248 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/768",
+
+    Annotations: 
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Marine wave energy transformation is a hydroelectric energy transformation that converts marine wave energy to electrical energy.",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "issue: https://github.com/OpenEnergyPlatform/ontology/issues/762
```
Here you used the `definition` property for a `term tracker item`.

sfluegel05 reviewed (commented):
Thanks for implementing this quite extensive issue.
A thing I am missing are the relations between the primary energies (C) and the transformation processes (B), i.e, B `has input` C.
For the power plants, you included the relation to the transformation process only for `marine tidal energy powerplant`. I personally have no preference on whether these axioms are needed, but we should keep them consistent.

l-emele commented:
The size of the issue and the PR is at the edge what is doable at one time. So thank you very much for thoroughly checking my implementation. 

Regarding B `has input` C: You're right, I forgot that. But shouldn't it be `has physical input` instead of `has input`?

sfluegel05 commented:
You're right. It should be `has physical input`.

l-emele commented:
I added now the `has physical input` relations.

For notice: I added `uses` relation between `hydro energy` and its subclasses and `water flow` and its subclasses. This is parallel with the structure between `wind energy` and `wind`.

sfluegel05 reviewed (approved):
Now we should have everything.
